### PR TITLE
Use UTF8String instead of Pointer_stringify

### PR DESCRIPTION
**Pull request by bas-d**
This is part of the fix for #196.

### CODE REVIEW

File: wrapper/build-wrappers.js [hunk 1]
```diff
@@ -104,7 +104,7 @@ function exportConstants(constSymbols) {
   exportsCode += "var constants_str = [" + keys.sort().join(", ") + "];\n";
   exportsCode += "for (var i = 0; i < constants_str.length; i++) {\n";
   exportsCode += '  var raw = libsodium["_" + constants_str[i].toLowerCase()];\n';
-  exportsCode += '  if (typeof raw === "function") exports[constants_str[i]] = libsodium.Pointer_stringify(raw());\n';
+  exportsCode += '  if (typeof raw === "function") exports[constants_str[i]] = libsodium.UTF8ToString(raw(),raw.length);\n';
```
Comment (jedisct1) [inline wrapper/build-wrappers.js]: Shouldn't it be `raw().length`?

File: wrapper/symbols/crypto_pwhash_scryptsalsa208sha256_str.json [hunk 2]
```diff
@@ -24,5 +24,5 @@
         ],
         "target": "libsodium._crypto_pwhash_scryptsalsa208sha256_str(hashed_password_address, password_address, password_length, 0, opsLimit, 0, memLimit) | 0",
         "assert_retval": [{"condition": "=== 0", "or_else_throw": "invalid usage"}],
-        "return": "libsodium.Pointer_stringify(hashed_password_address)"
+        "return": "libsodium.UTF8ToString(hashed_password_address, password_length)"
```
Comment (jedisct1) [inline wrapper/symbols/crypto_pwhash_scryptsalsa208sha256_str.json]: I don't think `password_length` is the length of the hash.

File: wrapper/symbols/crypto_pwhash_str.json [hunk 3]
```diff
@@ -24,5 +24,5 @@
         ],
         "target": "libsodium._crypto_pwhash_str(hashed_password_address, password_address, password_length, 0, opsLimit, 0, memLimit) | 0",
         "assert_retval": [{"condition": "=== 0", "or_else_throw": "invalid usage"}],
-        "return": "libsodium.Pointer_stringify(hashed_password_address)"
+        "return": "libsodium.UTF8ToString(hashed_password_address, password_length)"
```
Comment (jedisct1) [inline wrapper/symbols/crypto_pwhash_str.json]: Same.

File: wrapper/symbols/crypto_pwhash_scryptsalsa208sha256_str.json (hunk 2 above)
Comment (bas-d) [reply wrapper/symbols/crypto_pwhash_scryptsalsa208sha256_str.json]: You're probably right. Should it be `libsodium._crypto_pwhash_scryptsalsa208sha256_strbytes()`?

File: wrapper/symbols/crypto_pwhash_str.json (hunk 3 above)
Comment (bas-d) [reply wrapper/symbols/crypto_pwhash_str.json]: And `libsodium._crypto_pwhash_strbytes()` here?

File: wrapper/symbols/crypto_pwhash_scryptsalsa208sha256_str.json (hunk 2 above)
Comment (jedisct1) [reply wrapper/symbols/crypto_pwhash_scryptsalsa208sha256_str.json]: These strings are `\0`-terminated, so maybe we don't need the length at all :)

File: wrapper/symbols/crypto_pwhash_str.json (hunk 3 above)
Comment (jedisct1) [reply wrapper/symbols/crypto_pwhash_str.json]: Same here, we don't need the length.

File: wrapper/symbols/crypto_pwhash_scryptsalsa208sha256_str.json (hunk 2 above)
Comment (bas-d) [reply wrapper/symbols/crypto_pwhash_scryptsalsa208sha256_str.json]: Just read this on [emscripten](https://emscripten.org/docs/api_reference/preamble.js.html):
> N.B. mixing frequent uses of UTF8ToString() with and without maxBytesToRead may throw JS JIT optimizations off, so it is worth to consider consistently using one style or the other.

Not sure if it matters, but removed them from all the calls :)

**Comment by bas-d** [comment]
I think the difference in `drop_console` is caused by side effects that originate in emscripten. Probably best to fix it properly there before merging these changes. Should work with either `drop_console`  true or false. See https://github.com/emscripten-core/emscripten/pull/8558

**Comment by bas-d** [comment]
Ok turns out `drop_console` breaks things because of the way emscripten allows rebinding it with `Module.print` and `Module.printErr`. It can be easily fixed by explicitly rebinding these in `emscripten.sh` of the libsodium repo. I'll submit a PR there. I think we  need this one as well because it gives a deprecation warning for `Pointer_stringify` otherwise.

**Comment by jmtucker7** [comment]
Hi!  Are the issues with this commit resolved at this point?  It looks like the libsodium.js PR that came from this discussion has been merged.

**Pull request merged**